doc/how_to_guides/using_ompl_constrained_planning/ompl_constrained_planning.rst

```diff
@@ -0,0 +1,293 @@
+Using OMPL Constrained Planning
+===============================
+
+.. image:: ompl_constrained_planning_header.png
+   :width: 600px
+
+This tutorial shows you how to use OMPL's `Constrained planning capabilities <http://ompl.kavrakilab.org/constrainedPlanning.html>`_ from MoveIt. To illustrate the capabilities of this planner, four planning problems are solved with different types of path constraints. It will be easier to follow if you have read through the example on how to use the  :doc:`Move Group interface </doc/examples/move_group_interface/move_group_interface_tutorial>`.
+
+TODO(seng) Verify this video applies to MoveIt2. For a short walkthrough of running the tutorial, you can watch `this video <https://youtu.be/RkPydgtIq-M>`_.
+
+When to use this Planner
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The interface currently only supports a single position or orientation constraint on any link of the robot. The planning approach provides an alternative for the `enforce_joint_model_state_space <../../examples/ompl_interface/ompl_interface_tutorial.html#enforce-planning-in-joint-space>`_ option. It is expected to be most valuable for constraints regions that have a small (or zero) volume in Cartesian space, where rejection sampling does not always work. For example, this planner can be used to constrain the robot end-effector on a plane or along a line.
```

Comment:
Just to be sure  `on any link` or `on any joint`?

Reply:
I think it should be link. We enforce a constraint on the link such that the link cannot leave a constrained area / orientation, but the joints are free to move in any way that maintains that constraint. Unfortunately, the example uses `panda_link8` for the constrained link which has no geometry, which might look confusing.